api/src/main/java/io/minio/encryption/EncryptionMaterials.java

```diff
@@ -0,0 +1,132 @@
+/*
+ * Minio Java SDK for Amazon S3 Compatible Cloud Storage,
+ * (C) 2017 Minio, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.minio.encryption;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.security.InvalidAlgorithmParameterException;
+import java.security.InvalidKeyException;
+import java.security.NoSuchAlgorithmException;
+import java.security.SecureRandom;
+import java.util.HashMap;
+import java.util.Map;
+
+import javax.crypto.BadPaddingException;
+import javax.crypto.Cipher;
+import javax.crypto.CipherInputStream;
+import javax.crypto.IllegalBlockSizeException;
+import javax.crypto.KeyGenerator;
+import javax.crypto.NoSuchPaddingException;
+import javax.crypto.SecretKey;
+import javax.crypto.spec.IvParameterSpec;
+import javax.crypto.spec.SecretKeySpec;
+
+import org.apache.commons.codec.binary.Base64;
+
+import io.minio.ObjectStat;
+
+public class EncryptionMaterials {
+
+	private final static String cipherModeString = "AES/CBC/PKCS5Padding";
+	private final byte AESBlockSize = 16;
+	private Map<String, String> headerMap = new HashMap<>();
+	
+	public EncryptionMaterials(SecretKey symmetricMasterKey) throws 
+	InvalidKeyException, NoSuchAlgorithmException, 
+	NoSuchPaddingException, InvalidAlgorithmParameterException{
+		
+		// Setup symmetric keys and related ciphers
+		SymmetricKey.setupSymmetricKeys(symmetricMasterKey);
+		
+	}
+	
+	public CipherInputStream decryptInputStream(InputStream encryptedInputStream, ObjectStat metadata) throws 
```

Comment:
I could think of eliminating Object/HTTP concepts here.  Is it possible to make it as generic package where it could accept whatever info required to encrypt/decrypt Input/Output streams?